tensorflow_hub/resolver_test.py

```diff
@@ -250,6 +250,39 @@ def fake_download_fn_with_rogue_behavior(handle, tmp_dir):
         "Downloader Hostname: %s .PID:%d." % (re.escape(socket.gethostname()),
                                               os.getpid()))
 
+  def testModuleDownloadedWhenEmptyFolderExists(self):
+    # Simulate the case when a module is cached in /tmp/module_dir but module
+    # files inside the folder are deleted. In this case, the download should
+    # still be conducted.
+    module_dir = os.path.join(self.get_temp_dir(), "module")
+    def fake_download_fn(handle, tmp_dir):
+      del handle, tmp_dir
+      tf_v1.gfile.MakeDirs(module_dir)
+      tf_utils.atomic_write_string_to_file(
+          os.path.join(module_dir, "file"), "content", False)
+
+    # Delete existing folder and create an empty one.
+    if tf_v1.gfile.Exists(module_dir):
```

Comment:
That shouldn't be the case sinec get_temp_dir() should return a unique directory. May be just self.assertFalse(tf_v1.gfile.Exists(...))